### ISSUE TITLE
Go packages in the layer and meta-golang layer

### ISSUE THREAD

**Issue by roman-khimov** [morphis/meta-snappy#1]
At the moment, this layer brings its own cross Go recipes along with some Go package recipes. This conflicts with [meta-golang][1] if one tries to use both. IMO, it would be better to use that layer as a dependency, since it's well-maintained, has recipes for Go itself (and nice golang class to ease cross-compilation of Go recipes) and has some of snapd package dependencies already.

Had this been considered? Are there any plans to do that?

[1]: https://github.com/madisongh/meta-golang

**Comment by morphis** [comment]
Yes, this was considered but there was no time yet to switch to the go toolchain which is provided by meta-golang. If you want to contribute to get us ported to base things on meta-golang patches are very welcome. Otherwise I will see if I find some time in the coming weeks.

**Comment by morphis** [comment]
We're coming closer to this. Now that all golang-* recipes are dropped in favor of a vendorized tree for snapd I will look into using meta-golang as a base for the Go compiler next.

**Comment by roman-khimov** [comment]
Things are changing quite fast and today I think you should target oe-core Go infrastructure. I haven't played with it myself, but as it is something in oe-core it's gonna be way more widely used than separate layer.

**Comment by morphis** [comment]
@roman-khimov That is good to hear! Will play a little bit with that.
